src/Java/gtPlusPlus/xmod/gregtech/recipes/GregtechRecipeAdder.java

```diff
@@ -1731,8 +1758,77 @@ public boolean addFuelForRTG(ItemStack aFuelPellet, int aFuelDays, int aVoltage)
 	}
 
 
+	@Override
+	public boolean addColdTrapRecipe(int aCircuit, ItemStack aInput, FluidStack aFluidInput, ItemStack[] aOutputs, int[] aChances, FluidStack aFluidOutput, int aTime, int aEU) {
+		GTPP_Recipe aRecipe = new GTPP_Recipe(
+				false,
+				new ItemStack[] {CI.getNumberedAdvancedCircuit(aCircuit), aInput},
+				aOutputs,
+				null,
+				aChances,
+				new FluidStack[] {aFluidInput},
+				new FluidStack[] {aFluidOutput},
+				aTime,
+				aEU,
+				0);		
 
+		int aSize = GTPP_Recipe_Map.sColdTrapRecipes.mRecipeList.size();
+		GTPP_Recipe_Map.sColdTrapRecipes.add(aRecipe);
+		return GTPP_Recipe_Map.sColdTrapRecipes.mRecipeList.size() > aSize;
+	}
+
+
+	@Override
+	public boolean addReactorProcessingUnitRecipe(ItemStack aInput1, ItemStack aInput2, FluidStack aFluidInput, ItemStack[] aOutputs, int[] aChances, FluidStack aFluidOutput, int aTime, int aEU) {
+		GTPP_Recipe aRecipe = new GTPP_Recipe(
+				false,
+				new ItemStack[] {aInput1, aInput2},
+				aOutputs,
+				null,
+				aChances,
+				new FluidStack[] {aFluidInput},
+				new FluidStack[] {aFluidOutput},
+				aTime,
+				aEU,
+				0);		
+
+		int aSize = GTPP_Recipe_Map.sReactorProcessingUnitRecipes.mRecipeList.size();
+		GTPP_Recipe_Map.sReactorProcessingUnitRecipes.add(aRecipe);
+		return GTPP_Recipe_Map.sReactorProcessingUnitRecipes.mRecipeList.size() > aSize;
+	}
+
+
+    @Override
+    public boolean addFluidHeaterRecipe(ItemStack aInput, FluidStack aFluidInput, FluidStack aOutput, int aDuration, int aEUt) {
+        if ((aInput == null && aFluidInput == null) || (aOutput == null)) {
+            return false;
+        }
+        GT_Recipe.GT_Recipe_Map.sFluidHeaterRecipes.addRecipe(true, new ItemStack[]{aInput}, null, null, new FluidStack[]{aFluidInput}, new FluidStack[]{aOutput}, aDuration, aEUt, 0);
+        return true;
+    }
+
+
+    @Override
+    public boolean addVacuumFreezerRecipe(ItemStack aInput, ItemStack aOutput, int aDuration, int aEUt) {
+        if ((aInput == null) || (aOutput == null)) {
+            return false;
+        }        
+        GTPP_Recipe aRecipe = new GTPP_Recipe(
+				false,
+				new ItemStack[] {aInput},
+				new ItemStack[] {aOutput},
+				null,
+				new int[] {10000},
+				new FluidStack[] {},
+				new FluidStack[] {},
+				aDuration,
+				aEUt,
+				0);		
 
+		int aSize = GT_Recipe_Map.sVacuumRecipes.mRecipeList.size();
+		GT_Recipe_Map.sVacuumRecipes.add(aRecipe);
```

Comment:
You are not checking for collisions here - can that be problematic?

Reply:
It's only used for my hot ingots.